src/config.rs

```diff
@@ -0,0 +1,83 @@
+use std::fs::File;
+use std::io::{Read, Write};
+use rocket::config::{Config, Environment, Limits};
+
+use dirs;
+
+#[derive(Serialize,Deserialize)]
+pub struct AWConfig {
+    #[serde(default = "default_address")]
+    address: String,
+    #[serde(default = "default_port")]
+    port: u16,
+    // TODO: add CORS origins once that is properly supported
+}
+
+impl Default for AWConfig {
+    fn default() -> AWConfig {
+        AWConfig {
+            address: default_address(),
+            port: default_port(),
+        }
+    }
+}
+
+impl AWConfig {
+    pub fn to_rocket_config(&self) -> rocket::Config {
+        let env = Environment::active().expect("Failed to get current environment");
+        // Needed for bucket imports
+        let limits = Limits::new()
+            .limit("json", 1*1000*1000*1000);
+
+        Config::build(env)
+            .address(self.address.clone())
+            .port(self.port)
+            .limits(limits)
+            .finalize().unwrap()
+    }
+}
+
+fn default_address() -> String {
+    "127.0.0.1".to_string()
+}
+
+fn default_port() -> u16 {
+    match Environment::active().expect("Failed to get current environment") {
+        Environment::Production => 5600,
+        Environment::Development => 5666,
+        Environment::Staging => panic!("Staging environment not supported"),
+    }
+}
+
+pub fn get_config() -> AWConfig {
+    let mut config_path = dirs::get_config_dir().unwrap();
+    match Environment::active().expect("Failed to get current environment") {
+        Environment::Production => config_path.push("config.toml"),
+        Environment::Development => config_path.push("config-testing.toml"),
+        Environment::Staging => panic!("Staging environment not supported"),
+    };
+
+    if !config_path.is_file() {
+        debug!("Writing default config at {:?}", config_path);
```

Comment:
I don't think we should write a default configuration. The defaults should be loaded in memory, optionally overloaded by whatever is specified in the config file.

I know this is not how we do it in the Python code, but I've been thinking a while that we've been doing it wrong.

Reply:
Good point.
On the other hand it is nice to have all options visible in the config.
Not sure what the best option would be to be honest, both has its flaws.

Reply:
True, although the options should be documented anyway. 

Perhaps we could add the default options to the config, but commented out?

Reply:
Fixed